programs/fileio.c

```diff
@@ -1655,6 +1649,28 @@ static unsigned long long FIO_getLargestFileSize(const char** inFileNames, unsig
     return maxFileSize;
 }
 
+static const char*
+FIO_determineMirroredDestName(const char* srcFileName, const char* outDirName, const char* suffix)
+{
+    char *srcFileNameCopy = NULL, *srcDirName = NULL, *finalOutDirName = NULL;
+    const char *dest = NULL;
+
+    if (!UTIL_isFileNameValidForMirroredOutput(srcFileName))
+        return NULL;
+
+    srcFileNameCopy = strdup(srcFileName);
```

Comment:
This fixes the `C90` declaration - statement order,
but `strdup()` may still fail and return `NULL` as a consequence.

Reply:
done. will do a exit() upon strdup or malloc failure.